boot/nuttx/src/flash_map_backend/flash_map_backend.c

```diff
@@ -552,7 +552,7 @@ int flash_area_erase(const struct flash_area *fa, uint32_t off, uint32_t len)
       return ERROR;
     }
 
-  memset(buffer, erase_val, sizeof(buffer));
+  memset(buffer, erase_val, sizeof(*buffer));
```

Comment:
Not OK using `sizeof(*buffer)`, should probably be `memset(buffer, erase_val, sector_size);`

Reply:
**Right**. With `memset(buffer, erase_val, sector_size)` compiles correctly.

Reply:
Indeed, the solution proposed by @utzig should be the correct one.